src/audio/pipeline/pipeline-graph.c

```diff
@@ -423,29 +423,35 @@ int pipeline_for_each_comp(struct comp_dev *current,
 }
 
 /* visit connected pipeline to find the dai comp */
-struct comp_dev *pipeline_get_dai_comp(uint32_t pipeline_id)
+struct comp_dev *pipeline_get_dai_comp(uint32_t pipeline_id, int dir)
 {
-	struct ipc_comp_dev *sink;
+	struct ipc_comp_dev *crt;
 	struct ipc *ipc = ipc_get();
 
-	sink = ipc_get_ppl_sink_comp(ipc, pipeline_id);
-	while (sink) {
+	crt = ipc_get_ppl_comp(ipc, pipeline_id, dir);
+	while (crt) {
```

Comment:
this could be
```
for (crt = ipc_get_ppl_comp(ipc, pipeline_id, dir);
	crt;
	crt = ipc_get_ppl_comp(ipc, comp->pipeline->pipeline_id, dir))
```

Reply:
@lyakh i think this is a matter of taste. I would prefer to keep the existing while() loop because it was already there. Eventually, if one considers `for` approach to be easier to read we can send a patch later.